src/libs/TransactionUtils/index.ts

```diff
@@ -163,7 +163,7 @@ function buildOptimisticTransaction(
         created: created || DateUtils.getDBTime(),
         pendingAction: CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD,
         receipt: receipt?.source ? {source: receipt.source, state: receipt.state ?? CONST.IOU.RECEIPT_STATE.SCANREADY} : {},
-        filename: (receipt?.source ?? receipt?.name ?? filename).toString(),
+        filename: (receipt?.source ? receipt?.name ?? filename : filename).toString(),
```

Comment:
Don't we need to also check `receipt?.name` in the first part of the ternary?

Reply:
Oh, if there's no `receipt.name` it would use the `filename` though.

Reply:
I think in the title it says use receipt name if receipt source. So I think it uses receipt.name if receipt.source exists and receipt.name exists. Otherwise, it falls back on filename. Is that right @neil-marcellini?

Reply:
Maybe it would be easier to read if it's written as:

`(receipt?.source && receipt?.name) ? receipt?.name : filename`